src/Umbraco.Web/Mvc/UmbracoViewPageOfTModel.cs

```diff
@@ -105,18 +107,18 @@ protected UmbracoViewPage()
             : this(
                 Current.Factory.GetInstance<ServiceContext>(),
                 Current.Factory.GetInstance<AppCaches>(),
-                Current.Factory.GetInstance<IGlobalSettings>(),
-                Current.Factory.GetInstance<IContentSettings>()
+                Current.Factory.GetInstance<IOptions<GlobalSettings>>(),
+                Current.Factory.GetInstance<IOptions<ContentSettings>>()
             )
         {
         }
 
-        protected UmbracoViewPage(ServiceContext services, AppCaches appCaches, IGlobalSettings globalSettings, IContentSettings contentSettings)
+        protected UmbracoViewPage(ServiceContext services, AppCaches appCaches, IOptions<GlobalSettings> globalSettings, IOptions<ContentSettings> contentSettings)
         {
             Services = services;
             AppCaches = appCaches;
-            _globalSettings = globalSettings ?? throw new ArgumentNullException(nameof(globalSettings));
-            _contentSettings = contentSettings ?? throw new ArgumentNullException(nameof(contentSettings));
+            _globalSettings = globalSettings.Value;
```

Comment:
Why are these not required anymore, i.e. why was the exception removed?

Reply:
This is part of `Umbraco.Web`, but I introduced the null check. Generally we don't do lots of null checks in ctors because it should only be used by the DI container

Reply:
In some cases wouldn't the null checks help with unit tests?